### PR TITLE
Color picker

### PR DESCRIPTION
**Pull request by HaleyWardo**
**Changes**
- Added color picker
 
https://user-images.githubusercontent.com/32574227/122842556-a007ec80-d2cb-11eb-8b85-441765839f12.mov

### CODE REVIEW

**Comment by HaleyWardo** [comment]
I paused on adding this because I was running into some performance issues. Instead, I decided to put this component into PHC. After adding it in this morning, @ynotdraw pointed out that maybe the performance issue was caused because the story was rendering a lot of `ColorPicker`s... which was the case. 🤦‍♀️  Once this is approved I will update this [PR](https://github.com/lifeomic/phc-ui/pull/1229) to just use this component.

**Review by benkraus — APPROVED** [review]

File: src/components/ColorPicker/ColorPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+              <ButtonUnstyled
+                aria-label="Pick color"
+                disabled={disabled || readOnly}
+              >
+                <Color className={classes.colorPosition} color={colorValue} />
+              </ButtonUnstyled>
+            }
+            aria-label="Color Picker"
+          >
+            {({ popover }: { popover: any }) => (
```
Comment (ynotdraw) [inline src/components/ColorPicker/ColorPicker.tsx:406]: I could be wrong, but I think we export a type for the render props here:

```suggestion
            {({ popover }: PopoverRenderProps) => (
```

File: src/components/ColorPicker/ColorPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+  errorFor,
+  FormErrorMessage,
+  FormHelpMessage,
+  helpFor,
+} from '../_private/forms';
+import { ButtonUnstyled } from '../ButtonUnstyled';
+import { generateUniqueId } from '../_private/UniqueId';
+import { GetClasses } from '../../typeUtils';
+import { makeStyles, useTheme } from '../../styles';
+import { Popover, PopoverItem, PopoverList } from '../Popover';
```
Comment (ynotdraw) [inline src/components/ColorPicker/ColorPicker.tsx:15]: In combination with below:

```suggestion
import { Popover, PopoverItem, PopoverList, PopoverRenderProps } from '../Popover';
```

File: src/components/ColorPicker/ColorPicker.tsx (hunk 1 above)
Comment (HaleyWardo) [reply src/components/ColorPicker/ColorPicker.tsx]: Ooo for some reason I only tried `PopoverReturnProps`. Thank you! 🙏

**Review by xdumaine — APPROVED** [review]

**Review by ynotdraw — APPROVED** [review]

**Pull request merged**